File: product_array.py
from functools import reduce
def product_array(numbers):
    new_arr = []
    count = len(numbers)
    while count > 0:
        first = numbers.pop(0)
        new_arr.append(reduce(lambda x,y : x * y, numbers))
        numbers.append(first)
        count -= 1
    return new_arr

File: productArray.js
function productArray(numbers){
    let newArr = [];
    let count = numbers.length;
    while (count > 0){
      let first = numbers.shift();
      newArr.push(numbers.reduce((a,b) => a * b, 1));
      numbers.push(first);
       count--;
    }
   return newArr;
  }